zappa/cli.py

```diff
@@ -917,31 +931,29 @@ def update(self, source_zip=None, no_upload=False):
 
         # Register the Lambda function with that zip as the source
         # You'll also need to define the path to your lambda_handler code.
+        kwargs = dict(
+            bucket=self.s3_bucket_name,
+            function_name=self.lambda_name,
+            num_revisions=self.num_retained_versions
+        )
+        if self.use_alb:
+            kwargs['alias_name'] = self.alb_lambda_alias
         if source_zip and source_zip.startswith('s3://'):
             bucket, key_name = parse_s3_url(source_zip)
-            self.lambda_arn = self.zappa.update_lambda_function(
-                                            bucket,
-                                            self.lambda_name,
-                                            key_name,
-                                            num_revisions=self.num_retained_versions
-                                        )
+            kwargs = kwargs.update(dict(
+                bucket=bucket,
+                s3_key=key_name
+            ))
+            self.lambda_arn = self.zappa.update_lambda_function(**kwargs)
```

Comment:
TODO: [Simplify this by always creating an alias, even without ALB support.](https://github.com/youcandanch/Zappa/commit/9d13780daa1f5cf22f48ebc4b9e25ae5f79ba83e#r32557878)